### ISSUE TITLE
V1 Total Cost and Total Reward are subject to overflow

### ISSUE THREAD

**Issue by MitchTurner** [FuelLabs/fuel-core#2147]
We either need to use bigger integers (e.g. `u128`) or trim the totals down occasionally

**Comment by Nithin-Varma** [comment]
yes, use bigger integers and always use a library for better checks and asserts, similar to openzeppelin safemath.sol for solidity contracts.
if possible, please provide me the exact file, i want to look into it once.
